client/src/test/java/org/opensearch/ml/client/MachineLearningNodeClientTest.java

```diff
@@ -676,6 +689,84 @@ public void createConnector() {
+        MCorrModelTensors modelTensors = MCorrModelTensors.builder().mCorrModelTensors(mlModelTensors).build();
+        outputs.add(modelTensors);
+        metricsCorrelationOutput = MetricsCorrelationOutput.builder().modelOutput(outputs).build();
+
+        doAnswer(invocation -> {
+            ActionListener<MLExecuteTaskResponse> actionListener = invocation.getArgument(2);
+            MLExecuteTaskResponse output = new MLExecuteTaskResponse(FunctionName.METRICS_CORRELATION, metricsCorrelationOutput);
+            actionListener.onResponse(output);
+            return null;
+        }).when(client).execute(eq(MLExecuteTaskAction.INSTANCE), any(), any());
+
+        ArgumentCaptor<MLExecuteTaskResponse> argumentCaptor = ArgumentCaptor.forClass(MLExecuteTaskResponse.class);
+
+        List<float[]> inputData = new ArrayList<>(
+            Arrays
+                .asList(
+                    new float[] {
+                        0.89451003f,
+                        4.2006273f,
+                        0.3697659f,
+                        2.2458954f,
+                        -4.671612f,
+                        -1.5076426f,
+                        1.635445f,
+                        -1.1394824f,
+                        -0.7503817f,
+                        0.98424894f,
+                        -0.38896716f,
+                        1.0328646f,
+                        1.9543738f,
+                        -0.5236269f,
+                        0.14298044f,
+                        3.2963762f,
+                        8.1641035f,
+                        5.717064f,
+                        7.4869685f,
+                        2.5987444f,
+                        11.018798f,
+                        9.151356f,
+                        5.7354255f,
+                        6.862203f,
+                        3.0524514f,
+                        4.431755f,
+                        5.1481285f,
+                        7.9548607f,
+                        7.4519925f,
+                        6.09533f,
+                        7.634116f,
+                        8.898271f,
+                        3.898491f,
+                        9.447067f,
+                        8.197385f,
+                        5.8284273f,
+                        5.804283f,
+                        7.089733f,
+                        9.140584f }
+                )
+        );
+        Input metricsCorrelationInput = MetricsCorrelationInput.builder().inputData(inputData).build();
+
+        machineLearningNodeClient.execute(FunctionName.METRICS_CORRELATION, metricsCorrelationInput, executeTaskResponseActionListener);
+
+        verify(client).execute(eq(MLExecuteTaskAction.INSTANCE), isA(MLExecuteTaskRequest.class), any());
+        verify(executeTaskResponseActionListener).onResponse(argumentCaptor.capture());
+        assertEquals(FunctionName.METRICS_CORRELATION, argumentCaptor.getValue().getFunctionName());
```

Comment:
can we assert the output too?

Reply:
I added this additional assert cause you can never have too many tests but test coverage of this file is already at 100% methods and 100% lines:) 

Asserting anything specific related to metrics correlation algorithm should be covered when testing the algorithm unit itself, like in this file - https://github.com/opensearch-project/ml-commons/blob/main/ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelationTest.java